### PR TITLE
make sure logic is included in template

### PR DESCRIPTION
**Pull request by GeorgeGoodall**
Ticket: https://trello.com/c/SmInt8Gk/3209-bug-previous-next-pagination-buttons-always-showing-on-publish-tool

Testing for this falls under the requestData.js class testing. which is currently a tech debt ticket. Darren recently moved this from todo next into the backlog??? https://trello.com/c/F6hbYkyp/3196-improve-testing-surrounding-the-request-and-response-data-classes?search_id=38c93384-7a68-4306-8d26-b4c5ca490a47

I'll speak to him today and see if we can get this moved back to todo

### CODE REVIEW

**Review by cpcundill — CHANGES_REQUESTED** [review]
Looks like we need to add a test too so this doesn't break in future

File: test/unit/paginationTemplateTests.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    })
+
+    it('correctly renders the pagination when viewing the first page', () => {
```
Comment (cpcundill) [inline test/unit/paginationTemplateTests.js:59]: Looks like this should be when viewing the _last_ page?

File: test/unit/paginationTemplateTests.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      const pagination = {
+        nextPage: undefined,
+        previousPage: 9,
+        items: [
+          { number: 1, href: '/results/test/0', current: false },
+          { ellipsis: true, href: '#' },
+          { number: 8, href: '/results/test/7', current: false },
+          { number: 9, href: '/results/test/9', current: false },
+          { number: 10, href: '/results/test/9', current: true }
```
Comment (cpcundill) [inline test/unit/paginationTemplateTests.js:68]: This link should be `/results/test/10` non?

File: test/unit/requestData.test.js [hunk 3]
```diff
@@ -1,16 +1,105 @@
-import { pagination } from '../../src/models/requestData'
+import RequestData, { pagination } from '../../src/models/requestData'
 import { describe, it, expect } from 'vitest'
 
-// // Tech Debt: we should write some tests around the requestData.js file
```
Comment (cpcundill) [inline test/unit/requestData.test.js:4]: Nice to see Tech Debt being paid down

**Review by cpcundill — CHANGES_REQUESTED** [review]

File: test/unit/paginationTemplateTests.js (hunk 1 above)
Comment (GeorgeGoodall) [reply test/unit/paginationTemplateTests.js]: good spot ta

File: test/unit/paginationTemplateTests.js (hunk 2 above)
Comment (GeorgeGoodall) [reply test/unit/paginationTemplateTests.js:68]: line 67 or page 9 needs to actually point to /results/test/8 (now fixed)

the links are 0 indexed while the page numbers are one indexed. 

agree this should be amended but the team is eager for me to move on, so perhaps we leave this for now until we get a push from design to change this

**Pull request merged**